### PR TITLE
Wire NDC implementation end to end

### PR DESCRIPTION
**Pull request by wxing1292**
* Refactor replication queue, add handling of NDC replication task
* Fix issue that branch token should be set before task refreshment during mutable state rebuild
* Fix dead lock issue when performing NDC workflow reset
* Reorder task ID generation in ConflictResolveWorkflowExecution, ResetWorkflowExecution, making sure current workflow tasks are handled before new workflow tasks
* Add NDC handling for continue as new workflow in state builder, history engine, workflow resetter
* Fix cassandra schema

### CODE REVIEW

File: service/history/shardContext.go [hunk 1]
```diff
@@ -698,16 +698,6 @@ func (s *shardContextImpl) ConflictResolveWorkflowExecution(request *persistence
 	// assign IDs for the transfer/replication tasks
```
Comment (yux0) [inline service/history/shardContext.go]: Also move the comments?

File: service/history/historyEngine.go [hunk 2]
```diff
@@ -345,38 +345,51 @@ func (e *historyEngineImpl) registerDomainFailoverCallback() {
...
+) (mutableState, error) {
 
 	domainName := domainEntry.GetInfo().Name
-	if e.config.EnableEventsV2(domainName) && e.config.EnableNDC(domainName) {
-		return newMutableStateBuilderWithVersionHistories(
+	enableEventsV2 := e.config.EnableEventsV2(domainName)
+	enableNDC := e.config.EnableNDC(domainName)
+
+	var newMutableState mutableState
+	if enableEventsV2 && enableNDC {
```
Comment (yux0) [inline service/history/historyEngine.go]: So if event v2 is disable, it won't create mutable state with version histories?

File: service/history/replicatorQueueProcessor.go [hunk 3]
```diff
@@ -639,3 +548,323 @@ func (p *replicatorQueueProcessorImpl) getTasks(readLevel int64) (*replicator.Re
...
+}
+
+func (p *replicatorQueueProcessorImpl) getEventsBlob(
+	branchToken []byte,
+	firstEventID int64,
+	nextEventID int64,
+) (*shared.DataBlob, error) {
+
+	var eventBatchBlobs []*persistence.DataBlob
+	var pageToken []byte
```
Comment (yux0) [inline service/history/replicatorQueueProcessor.go]: no need to declare the pageToken here

File: service/history/historyEngine.go (hunk 2 above)
Comment (wxing1292) [reply service/history/historyEngine.go]: yes, how does it work without events v2?

**Pull request merged**